share/picongpu/tests/FieldAbsorber/lib/python/test/FieldAbsorber/validate.py

```diff
@@ -0,0 +1,176 @@
+"""
+This file is part of PIConGPU.
+
+Copyright 2023 PIConGPU contributors
+Authors: Klaus Steiniger
+License: GPLv3+
+"""
+
+import sys
+import numpy as np
+import openpmd_api as io
+
+
+def main(dataPath):
+    """ Evaluate the absorber performance
+
+        dataPath is supposed to point to the openPMD directory of the two
+        PIConGPU simulations where the files of the test simulation and
+        reference simulation are located.
+        Their naming follows the pattern simData_test_%T.bp and
+        simData_ref_%T.bp, respectively.
+
+        Return zero in case of sufficient absorption at the boundary and
+        one in case of too small absorption at the boundary.
+    """
+    series_test = io.Series(
+        dataPath + "/simData_test_%T.bp",
+        io.Access.read_only)
+    series_ref = io.Series(
+        dataPath + "/simData_ref_%T.bp",
+        io.Access.read_only)
+
+    """Read simulation parameters"""
+    # shape of a data set
+    Nt = len(series_test.iterations)
+    Nz, Ny, Nx = series_test.iterations[0].meshes["B"]["x"].shape
+    Nz_ref, Ny_ref, Nx_ref = series_ref.iterations[0].meshes["B"]["x"].shape
+
+    """ Define measurement position
+        Compared to sec. 7.11.1 'Current Source Radiating in an Unbounded
+        Two-Dimensional Region' of the Taflove book, point A corresponds
+        to an offset only along x, while point B corresponds to an offset
+        along x and y.
+    """
+    # Define the offset of the measurement point from the center of the
+    # conductor in cells.
+    # The conductor is located in the center of simulation domain.
+    # The offset is choosen such that the measurement point is located 2 cells
+    # in front of the PML in the small volume.
+    offset_cells = int(18)
+    x_offset = -offset_cells
+    y_offset = -offset_cells
+    z_offset = int(0)
+
+    # Calculate absolute position of measurement point in the small volume.
+    x_meas = Nx//2 + x_offset
+    y_meas = Ny//2 + y_offset
+    z_meas = Nz//2 + z_offset
+
+    # Calculate absolute position of measurement point in the reference sim.
+    x_meas_ref = Nx_ref//2 + x_offset
+    y_meas_ref = Ny_ref//2 + y_offset
+    z_meas_ref = Nz_ref//2 + z_offset
+
+    """Load data"""
+    times = np.zeros(Nt)
+
+    # create data containers.
+    # Explicitly provide data type in order to circumvent
+    # "RuntimeError: [Record_Component::load_chunk()]
+    #  Requires contiguous slab of memory."
+    # of openPMD
+    Ex = np.zeros(Nt, dtype=np.float32)
+    Ey = np.zeros(Nt, dtype=np.float32)
+    Ez = np.zeros(Nt, dtype=np.float32)
+
+    Ex_ref = np.zeros(Nt, dtype=np.float32)
+    Ey_ref = np.zeros(Nt, dtype=np.float32)
+    Ez_ref = np.zeros(Nt, dtype=np.float32)
+
+    i = 0
```

Comment:
please use `enuenumerate`  instead of a manual index `i`

Reply:
Done